_posts/2025-10-31-micrometer-prometheus-v1.adoc

```diff
@@ -0,0 +1,98 @@
+---
+layout: post
+title: 'Quarkus Micrometer using Prometheus client v1'
+date: 2025-10-31
+tags: micrometer prometheus
+synopsis: Micrometer has moved to use the Prometheus client v1. This post explains the migration path and required changes for applications.
+author: brunobat
+---
+
+== Quarkus Micrometer using Prometheus client v1
+
+Micrometer adopted Prometheus Client v1.x starting with version 1.13. However, Quarkus still uses Prometheus Client v0.x by default through the `quarkus-micrometer-registry-prometheus` extension which has been using the legacy `io.prometheus:simpleclient`.
+This is now considered *deprecated* and will not be part of a future Quarkus 4.x version (still no timeline).
+
+In parallel, we have introduced a new extension in Quarkiverse: https://docs.quarkiverse.io/quarkus-micrometer-registry/dev/micrometer-registry-prometheus-v1.html[`quarkus-micrometer-registry-prometheus-v1`], which integrates the new v1.x client.
+
+
+```xml
+<dependency>
+    <groupId>io.quarkiverse.micrometer.registry</groupId>
+    <artifactId>quarkus-micrometer-registry-prometheus-v1</artifactId>
+</dependency>
+```
+
+Since the new client introduces *breaking changes*, we’ve prepared a migration plan designed to reduce the impact on existing users.
+
+[INFO]
+====
+Some of the changes affect the names of some metrics. This requires queries and dashboards to be updated in order to continue working.
+Users should evaluate the impacts for their projects before moving to the new `quarkus-micrometer-registry-prometheus-v1` extension.
+====
+
+[WARNING]
+====
+The old and the new extension should not be used at the same time as the both require the `/q/metrics` endpoint.
+====
+
+=== Quarkus migration plan
+
+1. Keep current extension as is until Quarkus 4.0 (whenever that happens) so users can migrate at a time of their choice.
```

Comment:
1. You can keep using the deprecated extension until Quarkus 4.0, so you can do it at your own time.